16.0 mig commown factor out partner firstname portal

Honeyxilia opened the pull request.

Honeyxilia committed:
Create partner_firstname_portal module to replace name by firstname/lastname

Honeyxilia committed:
[test] Fixed request link with HttpCase method base_url

Honeyxilia committed:
[test] Removed useless inheritence

This test class doesn't use any method from RentalSaleOrderMixin, and it's not inherited by any classes.
As such, the RentalSaleOrderMixin is superfluous to the functionning of the test.

Honeyxilia committed:
[test] Adapted test to use Werkzeug client and methods

Beforehand, since we were using manually the HttpCase request methods, and encoding the parameters in the request url,
it meant the get_form/get_page methods weren't used, and it reduced lisibility.

To standardize the test with other similar tests and to use these previously unused methods, we now create a Werkzeug client and perform HTTP requests through it.

Honeyxilia committed:
[test] Adapted get_form method to use built-in form_values method

Honeyxilia committed:
Added field removal with t-if=False attribute

Using position="replace" is not handy, as it totally removes the targeted node which can be problematic if other modules rely on it.
As such, we simply insert a "t-if='False'" clause so that it doesn't get displayed.

Honeyxilia committed:
Adapted controller compatibility with new auth_signup implementation

- When the signup/password reset requests fetch the qcontext, it no longer copies all parameters,
  but only a select few keys defined in the SIGN_UP_REQUEST_PARAMS set, which does not contain
  the firstname/lastname keys, and is declared outside of a class meaning it can't be modified.
  As such, we copy the firstname/lastname values straight from the request parameters into the qcontext.

- Since the _prepare_signup_values method already checks the password and assigns a valid language,
  we only assign the name value in the qcontext constructed from the first and last name,
  then call the parent method.

Honeyxilia committed:
Renamed class and attribute to be more generic

fcayre commented on partner_firstname_portal/controllers/signup.py:
```diff
@@ -10,32 +10,24 @@
-        login, name and password. We compute a `name` because
-        `res_user` `_signup_create_user` methods asserts one is
-        supplied.
+        The original controller retrives every key in SIGN_UP_REQUEST_PARAMS,
```
Typos:
- message de commit : "only a select few keys" > "only select a few keys"
- docstring : "retrives" > "retrieves" ; "all key" > "all keys"

Honeyxilia replied:
Ok pour la docstring c:

Pour le message de commit, ici "select" est un adverbe, ce qui donne en français : "[...] il ne récupère pas tous les paramètres, mais seulement quelques clés sélectionnées"
-> https://www.wordreference.com/enfr/select voir la forme composée "select few"

fcayre commented on partner_firstname_portal/views/auth_signup.xml:
```diff
@@ -3,7 +3,7 @@
 
     <xpath expr="//div//input[@name='name']/ancestor::div[1]" position="attributes">
-      <attribute name="invisible">1</attribute>
+      <attribute name="t-if">False</attribute>
```
Je ne comprends pas ce commit, il n'y a pas d'usage de position="replace" ici. La façon canonique de l'OCA est bien d'utiliser invisible="1" pour rendre le champ invisible mais toujours présent pour les autres modules.

Juste dropper ce commit ?

Honeyxilia replied:
Non, car `invisible="1"` ne marche pas pour les éléments HTML, uniquement les champs de vues Odoo
Il faut ainsi utiliser t-if = False pour cacher un élément d'un template QWeb

Honeyxilia replied:
C'est par exemple utilisé dans `website_sale_payment_slimpay/views/address_template.xml`

fcayre replied:
+1

fcayre commented on partner_firstname_portal/tests/test_res_partner.py:
```diff
@@ -1,17 +1,17 @@
-from urllib.parse import urlencode
-
 from lxml import html
+from werkzeug.test import Client
```
Typo dans le message de commit : "Werkzerg" > "Werkzeug"

fcayre commented:
En passant une petite amélioration de la couverture de signup_retrieve_info serait bienvenue !

Honeyxilia committed:
[coverage] Changed signup_retrieve_info override to overload

Beforehand, we entirely rewrote the signup_restrieve_info method in order to replace the instructions
related to name, by instructions for firstname/lastname.
However, this does mean covering lines that are unrelated to the feature we're trying to add.

As such, we simply call the parent class method, then add firstname/lastname in the response.

Honeyxilia commented:
/ocabot merge patch

Honeyxilia commented:
/ocabot rebase

Honeyxilia commented:
Je ne comprends pas la source de ce bug mais je l'examinerai la semaine prochaine

Honeyxilia commented:
je pense avoir compris le souci  : en rebasant sur la version plus récente, j'ai déplacé du code en trop (en pensant que le fichier `commown/models/res_partner.py` entier pouvait être migré) qui assignait un pays de base à un•e partenaire...
je modifierai ça la semaine prochaine

Honeyxilia committed:
Moved code related to partner_firstname's inclusion in portal templates

Honeyxilia commented:
/ocabot merge patch
